CMakeLists.txt

```diff
@@ -139,11 +108,63 @@ set(CLP_FFI_JS_SRC_ZSTD
     src/submodules/zstd/lib/decompress/zstd_decompress.c
 )
 
-target_sources(
-    ${CLP_FFI_JS_BIN_NAME}
-    PRIVATE
+set(CLP_FFI_JS_SUPPORTED_ENVIRONMENTS
+    "node;worker"
+    CACHE INTERNAL
+    "List of supported environments."
+)
+
+foreach(env ${CLP_FFI_JS_SUPPORTED_ENVIRONMENTS})
+    set(CLP_FFI_JS_BIN_NAME "ClpFfiJs-${env}")
+    add_executable(${CLP_FFI_JS_BIN_NAME})
+
+    # Set up compile options
+    target_compile_features(${CLP_FFI_JS_BIN_NAME} PRIVATE cxx_std_20)
+    target_compile_definitions(${CLP_FFI_JS_BIN_NAME} PUBLIC SPDLOG_FMT_EXTERNAL=1)
+    target_compile_options(${CLP_FFI_JS_BIN_NAME} PRIVATE ${CLP_FFI_JS_COMMON_COMPILE_OPTIONS})
+
+    # Set up link options
+    target_link_libraries(${CLP_FFI_JS_BIN_NAME} PRIVATE embind)
+    set(CLP_FFI_JS_LINK_OPTIONS
+        ${CLP_FFI_JS_COMMON_LINK_OPTIONS}
```

Comment:
I understand the previous suggestion was to make an "extra link options" list and append to it as necessary. However, I'm not sure if the appended options could be undesirably carried onto the next `foreach` iterations. Instead of creating an "extra options" list, a "common options" list is created to merge with other options as necessary. Let me know if you see any drawbacks with this approach.